[v0.14.0.x] Update release notes for 0.14.0.2 (#3012)
dashpay/dash: doc/release-notes.md

Downgrade warning
-----------------

### Downgrade to a version < 0.14.0.0

Downgrading to a version smaller than 0.14 is not supported anymore as DIP8 has
activated on mainnet and testnet.

### Downgrade to versions 0.14.0.0 - 0.14.0.1

> Downgrading to older 0.14 releases is fully but is not

Reviewer: Downgrading to older 0.14 releases is fully supported but is not

Author: Thanks, fixed

recommended unless you have some serious issues with version 0.14.0.2.

Notable changes
===============

Performance improvements
------------------------
Slow startup times were observed in older versions. This was due to sub-optimal handling of old

> deterministic masternode lists which causes loading of too many lists into memory. This should be

Reviewer 2: Tense change seems technically wrong
```suggestion
deterministic masternode lists which caused loading too many lists into memory. This should be
```

Reviewer 3: This suggestion doesn't seem to flow as well as the previous... "... caused loading too many..." sounds weird. "...caused the loading of too many ..." sounds better IMO.

Reviewer 2: agreed ```suggestion
deterministic masternode lists which caused the loading of too many lists into memory. This should be
```

- [`0410259dd`](https://github.com/dashpay/dash/commit/0410259dd) Fix 2 common Travis failures which happen when Travis has network issues (#3003)
- [`8d763c144`](https://github.com/dashpay/dash/commit/8d763c144) Only load signingActiveQuorumCount + 1 quorums into cache (#3002)
- [`2dc1b06ec`](https://github.com/dashpay/dash/commit/2dc1b06ec) Remove skipped denom from the list on tx commit (#2997)
- [`dff2c851d`](https://github.com/dashpay/dash/commit/dff2c851d) Update manpages for 0.14.0.2 (#2999)
- [`fb6f0e04d`](https://github.com/dashpay/dash/commit/fb6f0e04d) Bump version to 0.14.0.2 and copy release notes (#2991)
- [`46c4f5844`](https://github.com/dashpay/dash/commit/46c4f5844) Use Travis stages instead of custom timeouts (#2948)
- [`49c37b82a`](https://github.com/dashpay/dash/commit/49c37b82a) Back off for 1m when connecting to quorum masternodes (#2975)
- [`c1f756fd9`](https://github.com/dashpay/dash/commit/c1f756fd9) Multiple speed optimizations for deterministic MN list handling (#2972)
- [`11699f540`](https://github.com/dashpay/dash/commit/11699f540) Process/keep messages/connections from PoSe-banned MNs (#2967)

> - [`c5415e746`](https://github.com/dashpay/dash/commit/c5415e746) Fix UI masternode list (#2966)

Reviewer 4: ```suggestion
- [`c5415e746`](https://github.com/dashpay/dash/commit/c5415e746) Fix UI masternode list (#2966)
- [`fb6f0e04d`](https://github.com/dashpay/dash/commit/fb6f0e04d) Bump version to 0.14.0.2 and copy release notes (#2991)
```

See detailed [set of changes](https://github.com/dashpay/dash/compare/v0.14.0.1...dashpay:v0.14.0.2).

- [`d2ff63e8d`](https://github.com/dashpay/dash/commit/d2ff63e8d) Use std::unique_ptr for mnList in CSimplifiedMNList (#3014)
- [`321bbf5af`](https://github.com/dashpay/dash/commit/321bbf5af) Fix excessive memory use when flushing chainstate and EvoDB (#3008)
- [`0410259dd`](https://github.com/dashpay/dash/commit/0410259dd) Fix 2 common Travis failures which happen when Travis has network issues (#3003)
- [`8d763c144`](https://github.com/dashpay/dash/commit/8d763c144) Only load signingActiveQuorumCount + 1 quorums into cache (#3002)
- [`2dc1b06ec`](https://github.com/dashpay/dash/commit/2dc1b06ec) Remove skipped denom from the list on tx commit (#2997)
- [`dff2c851d`](https://github.com/dashpay/dash/commit/dff2c851d) Update manpages for 0.14.0.2 (#2999)

> - [`fb6f0e04d`](https://github.com/dashpay/dash/commit/fb6f0e04d) Bump version to 0.14.0.2 and copy release notes (#2991)

Reviewer 4: Would prefer to preserve the order they appear in `git log` i.e. this should be the first one (at the bottom of the list)
```suggestion
```

Author: Strange, this is the order that I got from `git log --oneline --no-merges v0.14.0.1..v0.14.0.x`

Reviewer 4: Hmm... indeed. That's probably because #2989 wasn't rebased after #2991 was merged. `git log --oneline --graph` shows a bit more correct picture (imo) in terms of what was merged first.

Author: Ah ok, thanks. Will use `--graph` in the future :) Force pushed the fixed version

These release are considered obsolete. Old release notes can be found here:

> - [v0.14.1](https://github.com/dashpay/dash/blob/master/doc/release-notes/dash/release-notes-0.14.1.md) released May/31/2019

Reviewer 5: This link is broken

> - [v0.14.1](https://github.com/dashpay/dash/blob/master/doc/release-notes/dash/release-notes-0.14.1.md) released May/31/2019

Reviewer 4: yep, should be 0.14.0.1
```suggestion
- [v0.14.0.1](https://github.com/dashpay/dash/blob/master/doc/release-notes/dash/release-notes-0.14.0.1.md) released May/31/2019
```

Author: fixed

> - [v0.14.0.1](https://github.com/dashpay/dash/blob/master/doc/release-notes/dash/release-notes-0.14.0.1.md) released May/31/2019

Reviewer 2: have the old release notes been copied here yet? I'm not finding it

Author: They have been copied, but they are not in `master` yet. You can find them here: https://github.com/dashpay/dash/blob/v0.14.0.x/doc/release-notes/dash/release-notes-0.14.0.1.md

Reviewer 2: ahh, I was looking in develop
